datafusion/core/src/datasource/statistics.rs

```diff
@@ -39,8 +39,8 @@ pub async fn get_statistics_with_limit(
     file_schema: SchemaRef,
     limit: Option<usize>,
     collect_stats: bool,
-) -> Result<(Vec<PartitionedFile>, Statistics)> {
-    let mut result_files = vec![];
+) -> Result<(FileGroup, Statistics)> {
+    let mut result_files = FileGroup::default();
```

Comment:
What are the difference between the `statistics` in `FileGroup` and the returned value here? Do we need `statistics` in FileGroup, if so, could we add the returned `Statistics` inside `FileGroup`?

Reply:
Can we return `result_files.with_statistics(statistics)`? I didn't see other place where we call `with_statistics` the inner optional statistics is confusing to me

Reply:
@jayzhan211 , yeah, it's a todo task, I'll compute the `FileGroup`'s statistic in the method and use the `with_statistics` method to set statistic for `FileGroup` later.

Reply:
Fyi: https://github.com/apache/datafusion/pull/15432